README.md

```diff
@@ -23,23 +23,29 @@
 [bors-img]: https://bors.tech/images/badge_small.svg
 [bors-url]: https://app.bors.tech/repositories/32073
 
-A `DispatchedTuple` is like a dictionary, except
+## What are dispatched tuples?
 
- - the keys are **instances of types**
- - they are backed by tuples, so they are GPU-friendly
- - multiple keys are allowed (except for `DispatchedTupleSet`)
- - each unique key returns a tuple of values given by that key (in order)
+`DispatchedTuple`s are like immutable dictionaries (so, they're more like `NamedTuple`s) except that the keys are **instances of types**. Also, because `DispatchedTuple`s are backed by tuples, they are GPU-friendly.
```

Comment:
Ah I found this description much more insightful! I can see the use cases now and the name `DispatchedTuple` makes more sense now.